corgi/core/migrations/0040_auto_20221020_1057.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 3.2.15 on 2022-10-20 10:57
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("core", "0039_auto_20221013_0748"),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name="product",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_671dbe_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="productstream",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_e77387_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="productvariant",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_ccf4bd_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="productversion",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_c62cf2_idx"),
+        ),
+        migrations.RunSQL(
```

Comment:
Do you think we'd be able to adjust the query in get_latest_components to filter by component nodes which don't have parents instead of this type filter which is likely to change?

I just worry that if a new root type starts being used, we'll have to adjust and rebuild this index as well.

Reply:
using django-mptt the answer is probably no eg. structurally we will be better placed to have more generic indexes if we use something like ltree in the future.